sefaria/model/linker/tests/linker_test_utils.py

```diff
@@ -150,4 +152,48 @@ def print_spans(raw_ref: RawRef):
     print('\nInput:', raw_ref.text)
     print('Spans:')
     for i, part in enumerate(raw_ref.raw_ref_parts):
-        print(f'{i}) {part.text}')
\ No newline at end of file
+        print(f'{i}) {part.text}')
+
+
+_MISSING = object()
+
+
+def attach_context_mutations(target, mutations, *, append: bool = False):
```

Comment:
this function is very confusing. We should define the type of `target` and not guess what it is. I think `Ref` is fine.